tqdm/_tqdm.py

```diff
@@ -470,14 +476,11 @@ def _decr_instances(cls, instance):
                 for inst in cls._instances:
                     if inst.pos > instance.pos:
                         inst.pos -= 1
-            # Kill monitor if no instances are left
-            if not cls._instances and cls.monitor:
-                cls.monitor.exit()
-                try:
+                # Kill monitor if no instances are left
+                if not cls._instances and cls.monitor:
+                    cls.monitor.exit()
```

Comment:
@casperdcl why move the monitor exit inside the lock? It might work but did you try out? Because IIRC, I tried to put any action involving the monitor outside of locks, else the monitor might stuck everything (but I tell that by memory, I did not try this code).

Reply:
Related to comment below